File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
 
public class Solution {
    public List<List<int>> LevelOrder(TreeNode root) {
        List<List<int>> ans = new List<List<int>>();
        Queue<TreeNode> queue = new Queue<TreeNode>();

        if(root != null){
           queue.Enqueue(root);
        }

        while(queue.Count > 0){
            //get level lenght
            int levelLength = queue.Count;
            List<int> row = new List<int>();

            //Iterate through level
            for(int levelIndex=0; levelIndex < levelLength; ++levelIndex){
                 //Process Node
                 TreeNode curr = queue.Dequeue();
                 row.Add(curr.val);

                 //Get Children
                 if(curr.left != null){
                    queue.Enqueue(curr.left);
                 }
                 if(curr.right != null){
                    queue.Enqueue(curr.right);
                 }
            }

            ans.Add(row);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/path-sum/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        #base case
        if not root:
           return False

        #process current node, check if found root to leaf path
        targetSum -= root.val
        if root.left is None and root.right is None and targetSum == 0:
           return True

        #otherwise, recuse to children
        return self.hasPathSum(root.left, targetSum) or self.hasPathSum(root.right, targetSum)

         
      